### PR TITLE
[IPAM] function NetworkIsAvailable refactoring

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR changes how the NetworkIsAvailable internally work

### CODE REVIEW

File: pkg/ipam/core/node.go [hunk 1]
```diff
@@ -121,6 +121,31 @@ func networkRelease(prefix netip.Prefix, node *node) *netip.Prefix {
 	return result
 }
 
+func networkIsAvailable(prefix netip.Prefix, node *node) bool {
+	if node == nil {
+		return true
+	}
+
+	if node.prefix.Addr().Compare(prefix.Addr()) == 0 && node.prefix.Bits() == prefix.Bits() {
+		if node.left != nil && (node.left.left != nil || node.left.right != nil) {
```
Comment (claudiolor) [inline pkg/ipam/core/node.go:130]:
```suggestion
		if node.left != nil && node.left.isSplitted() {
```

File: pkg/ipam/core/node.go [hunk 2]
```diff
@@ -121,6 +121,31 @@ func networkRelease(prefix netip.Prefix, node *node) *netip.Prefix {
...
+			return false
+		}
+		if node.right != nil && (node.right.left != nil || node.right.right != nil) {
```
Comment (claudiolor) [inline pkg/ipam/core/node.go:133]:
```suggestion
		if node.right != nil && node.right.isSplitted() {
```

File: pkg/ipam/core/node.go [hunk 3]
```diff
@@ -121,6 +121,31 @@ func networkRelease(prefix netip.Prefix, node *node) *netip.Prefix {
...
+			return false
+		}
+		return !node.acquired
```
Comment (claudiolor) [inline pkg/ipam/core/node.go:136]:
```suggestion
                // If node is not splitted nor acquired, then network is available
		return !node.acquired
```

File: pkg/ipam/core/node.go (hunk 2 above)
Comment (cheina97) [reply pkg/ipam/core/node.go:133]: Same

**Review by fra98 — APPROVED** [review]

**Pull request merged**